src/contract/index.ts

```diff
@@ -158,18 +161,66 @@ export class VerifiedContract {
         else return [data]
     }
 
-    async callContract(functionName: string, ...args: any) {
+    async callContract(functionName: string, ...args: any):Promise<UserOperationEventEvent | SCResponse | null> {
         let res = <SCResponse>{};
         try {
             let options = []
             const totalArguments = args.length
-
+            // params to be used while performing gasless transaction with userop 
+            const params =  [...args];
+                if (totalArguments > 1){
+                    // Check if there is options and remove it because it should be removed last in arguments
+                    params.pop()
+                }
             if (totalArguments > 1) options = args.splice(-1)
             //console.log('options before', options);
 
             if (options == 0) options[0] = {}
             //console.log('*********', ...args)
             //console.log('options after', options);
+
+            // call the function using userop
+            // Detect options are parsed
+            // @ts-ignore
+            const isOptionsProvided = Object.keys(...options).length>0 && "gasPrice" in options[0];
+            if (isOptionsProvided && options[0].gasPrice===0) {
+                // Get the arguments for the contract after removing options
+                const funcParameters = params.length>0?params:undefined;
+
+                 // Define the kind of paymaster you want to use
+                    const paymasterContext = { type: "payg" };
+                    const paymasterMiddleware = Presets.Middleware.verifyingPaymaster(
+                        process.env.PAY_MASTER_URL,
+                        paymasterContext
+                    );
+
+                    const signer = new ethers.Wallet(process.env.API_KEY);
```

Comment:
@herberthk 

Why is a signer created again here ? The signer is already passed to the constructor

Reply:
It was just a naming conflict but this signer is different from what was passed in the constructor